feat!: add support for cas option for kv2 backend

bodymindarts opened the pull request:
This pull request adds support for vaults compare-and-swap feature. It contains breaking changes as the `kv2::set` signature is expanded with an additional `Option<SetSecretRequestOptions>` argument.

closes https://github.com/jmgilman/vaultrs/issues/33

jmgilman commented on src/lib.rs:
```diff
@@ -89,6 +89,7 @@
 //!     "secret",
 //!     "mysecret",
 //!     &secret,
+//!     None,
```
I believe this example also exists in the README.

bodymindarts commented:
I have addressed the comment regarding the `README` and added an alternative PR that is backwards compatible https://github.com/jmgilman/vaultrs/pull/35.
